### PR TITLE
Move all Storage implementations in storage.py

### PR DESCRIPTION
**Pull request by rgayon**
Some unused methods in AuFSStorage (such as FindContainerId) have been deleted too.
Storage has been renamed BaseStorage

### CODE REVIEW

File: docker_explorer/lib/storage.py [hunk 1]
```diff
@@ -23,15 +23,15 @@
...
+class BaseStorage(object):
   """This class provides tools to list and access containers metadata.
 
   Every method provided is agnostic of the implementation used (AuFS,
   btrfs, etc.).
   """
 
   def __init__(self, docker_directory='/var/lib/docker', docker_version=2):
-    """Initialized a Storage object.
+    """Initialized a BaseStorage object.
```
Comment (tomchop) [inline docker_explorer/lib/storage.py]: Initialize**s**

File: docker_explorer/lib/storage.py [hunk 2]
```diff
@@ -130,3 +142,137 @@ def Mount(self, container_object, mount_dir):
...
+    Args:
+      docker_directory (str): Path to the Docker root directory.
+      docker_version (int): Docker storage version.
+    """
+
+    super(AufsStorage, self).__init__(
+        docker_directory=docker_directory, docker_version=docker_version)
+
+  def MakeMountCommands(self, container_object, mount_dir):
+    """Generates the required shell commands to mount a container's ID.
```
Comment (tomchop) [inline docker_explorer/lib/storage.py]: To mount a container given its ID

File: docker_explorer/lib/storage.py [hunk 3]
```diff
@@ -130,3 +142,137 @@ def Mount(self, container_object, mount_dir):
...
+      lower (str): the path to the lower directory.
+
+    Returns:
+      str: the mount command option for thr 'lower' directory.
+    """
+    return os.path.join(
+        self.docker_directory, self.STORAGE_METHOD, lower.strip(), 'root')
+
+  def MakeMountCommands(self, container_object, mount_dir):
+    """Generates the required shell commands to mount a container's ID.
```
Comment (tomchop) [inline docker_explorer/lib/storage.py]: To mount a container given its ID

File: docker_explorer/lib/storage.py [hunk 4]
```diff
@@ -130,3 +142,137 @@ def Mount(self, container_object, mount_dir):
...
+    """
+    mount_id_path = os.path.join(
+        self.docker_directory, self.STORAGE_METHOD, container_object.mount_id)
+
+    with open(os.path.join(mount_id_path, self.LOWER_NAME)) as lower_fd:
+      lower_dir = self._BuildLowerLayers(lower_fd.read())
+    upper_dir = os.path.join(mount_id_path, self.UPPER_NAME)
+    work_dir = os.path.join(mount_id_path, 'work')
+
+    cmd = (
```
Comment (tomchop) [inline docker_explorer/lib/storage.py]: For better readability, I would split this into two lines, one with the string into a variable and the other one with `variable.format()`

File: docker_explorer/lib/storage.py [hunk 5]
```diff
@@ -130,3 +142,137 @@ def Mount(self, container_object, mount_dir):
...
+
+
+class Overlay2Storage(OverlayStorage):
+  """A specialization for Overlay2.
+
+  See a description at
+  https://docs.docker.com/storage/storagedriver/overlayfs-driver/#how-the-overlay2-driver-works.
+  """
+
+  STORAGE_METHOD = u'overlay2'
```
Comment (tomchop) [inline docker_explorer/lib/storage.py]: I don't see you manually using unicode literals in other strings

File: docker_explorer/lib/storage.py [hunk 6]
```diff
@@ -130,3 +142,137 @@ def Mount(self, container_object, mount_dir):
...
+    Args:
+      lower (str): the path to the lower directory.
+
+    Returns:
+      str: the mount command option for thr 'lower' directory.
+    """
+    # We need the full pathname to the lower directory.
+    # If multiple lower directories are stacked, we process each of them
+    # separately.
+    lower_dir = ':'.join([
```
Comment (tomchop) [inline docker_explorer/lib/storage.py]: This statement is not super clear, maybe separating the strip().split() into a different statement would make it more readable.

Also, I would include an example in the docstring of what this function does (or a reference to where the function is tested).

File: docker_explorer/lib/storage.py (hunk 1 above)
Comment (rgayon) [reply docker_explorer/lib/storage.py]: done

File: docker_explorer/lib/storage.py (hunk 2 above)
Comment (rgayon) [reply docker_explorer/lib/storage.py]: done

File: docker_explorer/lib/storage.py (hunk 3 above)
Comment (rgayon) [reply docker_explorer/lib/storage.py]: done

File: docker_explorer/lib/storage.py (hunk 5 above)
Comment (rgayon) [reply docker_explorer/lib/storage.py]: ='(

File: docker_explorer/lib/storage.py (hunk 4 above)
Comment (rgayon) [reply docker_explorer/lib/storage.py]: like so?

**Pull request merged**